src/gui/src/scriptWidget.cpp

```diff
@@ -208,9 +232,14 @@ void ScriptWidget::addLogToOutput(const QString& text, const QColor& color)
   addToOutput(text, color);
 }
 
-void ScriptWidget::addReportToOutput(const QString& text)
+void ScriptWidget::startReportTimer()
+{
+  report_timer_->start(report_display_interval);
+}
+
+void ScriptWidget::addMsgToReportBuffer(const QString& text)
 {
-  addToOutput(text, buffer_msg_);
+  report_buffer_ += text;
```

Comment:
this will also need a mutex here so different threads don't try to add at the same time.